node/activities-registration-redesign.md

```diff
@@ -0,0 +1,57 @@
+# Node.js Activities Registration
+
+## Problems with the current design
+
+- [#109](https://github.com/temporalio/sdk-node/issues/109) Activity identifier should not include the activity module name
+- `@activities` import hack
+
+  Currently we use the following to import Activity functions as stubs in a Workflow.
+
+  ```ts
+  import { myActivity } from '@activities/something';
+  ```
+
+  Issues:
+
+  - `@` makes it look like a namespaced package
+  - Too much _magic_
+  - Only way to register activities is by passing a directory for the Worker to scan
+
+## Proposal
+
+Another approach would be to reuse the concept from [`ExternalDependencies`](https://docs.temporal.io/docs/node/external-dependencies#example)
+
+`workflows/example.ts`
+
+```ts
+export interface ActivityInterface {
+  deposit(account: string, amount: number): Promise<void>;
+  withdraw(account: string, amount: number): Promise<void>;
+}
+
+// deposit and withdraw are typesafe stubs
+const { deposit, withdraw } = Context.activities<ActivityInterface>(activityOptions);
```

Comment:
We could add these functions, it might be useful in tests.
My main reason for not adding these functions to this proposal is that if you allow multiple registrations you have to deal with duplicates as opposed to accepting a single key-value object.

It also aligns nicely with stub creation, there's less room for error here since the activity functions are referenced in their entirety.

```ts
import * as activities from '../activities';
// workflow
const { deposit, withdraw } = Context.activities<typeof activities>(activityOptions);
// worker
const worker = await Worker.create({ activities });
```

Reply:
@bergundy Dealing with dupes is annoying I agree. The thing I still don't understand here is how do I specify different activity options for `deposit` and `withdraw`?

Reply:
@Sushisource I think Builder pattern can support variations such as this:

`Worker.withActivities(deposit, ao1).withActivities(withdraw, ao2).create()`

Reply:
@Sushisource
```ts
const { withdraw } = Context.configureActivities<typeof activities>(activityOptionsA);
const { deposit } = Context.configureActivities<typeof activities>(activityOptionsB);
```

Reply:
@bergundy Sorry that I keep confusing Context vs Worker functions...

However, if I understand the code above it looks like you're configuring some of the same activities twice.

Is it still possible to import the activities individually and configure them individually? If so, I would prefer that form in my own code.

Reply:
Note that activities aren't being imported here, we only use the types.
You can use `configureActivities` as many times as you want with any type params.

e.g.:

```ts
const { withdraw } = Context.configureActivities<MyActivitiesIfaceA>(activityOptionsA);
const { deposit } = Context.configureActivities<MyActivitiesIfaceB>(activityOptionsB);
``` 

You might understand more by reading the implementation: https://github.com/temporalio/sdk-node/blob/b97dd21aff3f5d1e5beb1fc6f4e71a04d761ac02/packages/workflow/src/workflow.ts#L383